flink-runtime/src/main/java/org/apache/flink/runtime/rpc/jobmaster/JobMaster.java

```diff
@@ -248,4 +374,40 @@ void finishResourceManagerRegistration(ResourceManagerGateway resourceManager, I
 	public boolean isConnected() {
 		return resourceManager != null;
 	}
+
+
+	/**
+	 * Cancel the current job and notify all listeners the job's cancellation.
+	 *
+	 * @param cause Cause for the cancelling.
+	 */
+	private void cancelAndClearEverything(Throwable cause) {
+		// currently, nothing to do here
+	}
+
+	// ------------------------------------------------------------------------
+	//  Utility classes
+	// ------------------------------------------------------------------------
+	private class JobMasterLeaderContender implements LeaderContender {
```

Comment:
I'm not really decided here: Maybe we could let the JobMaster directly implement the LeaderContender interface. Then we could get rid of one layer of indirection (basically the `grantJobMasterleadership`, the `revokeJobMasterLeadership` and `onJobMasterElectionError` methods. Would that make sense?